File: src/app/models/post.ts
export interface Post {
  id: number;
  title: string;
  content: string;
  createdAt: string;
  views: number;
  //   comments: Comment[]; 익명이라 댓글 수정/삭제가 애매해 우선 보류
  url?: string;
  thumbnailUrl?: string;
}

File: backend/models/post.py
from sqlalchemy import Column, Integer, String, Text, DateTime, JSON
from sqlalchemy.orm import relationship
from datetime import datetime
from .comment import Comment
from .base import Base

class Post(Base):
    __tablename__ = "post"

    id = Column(Integer, primary_key=True, autoincrement=True)
    title = Column(String(255), nullable=False)
    content = Column(Text, nullable=False)
    created_at = Column(DateTime, nullable=False, default=datetime.utcnow)
    views = Column(Integer, nullable=False, default=0)
    tags = Column(JSON)
    url = Column(String(500))
    thumbnail_url = Column(String(500))
    password_hash = Column(String(255), nullable=False)
    
    # Relationship
    comments = relationship(
        "Comment",
        cascade="all, delete-orphan",
        order_by=lambda: Comment.created_at
    )